src/CHIPBackend.cc

```diff
@@ -1445,6 +1450,17 @@ CHIPQueue *CHIPBackend::findQueue(CHIPQueue *ChipQueue) {
   return *QueueFound;
 }
 
+bool CHIPBackend::eraseProgram(const CHIPProgram *ToErase) {
+  auto ErasePos =
+      std::remove_if(ChipPrograms_.begin(), ChipPrograms_.end(),
+                     [&](const std::unique_ptr<CHIPProgram> &Prog) -> bool {
+                       return Prog.get() == ToErase;
+                     });
+  bool WasErased = ErasePos != ChipPrograms_.end();
+  ChipPrograms_.erase(ErasePos, ChipPrograms_.end());
```

Comment:
thread safe?

Reply:
I have a follow up PR for removing this. I'll transfer it here.